### PR TITLE
Move tracers to `.yaml` files

### PR DESCRIPTION
**Pull request by Jimbo4350**
Specifying the tracers on the command line is cumbersome and the consensus is to move to them to the configuration `.yaml` files. 

Relevant: #443

### CODE REVIEW

File: cardano-config/src/Cardano/Config/Types.hs [hunk 1]
```diff
@@ -324,38 +352,27 @@ instance FromJSON ViewMode where
 --   which verbosity to the log output.
 data TraceOptions = TraceOptions
   { traceVerbosity :: !TracingVerbosity
+  , traceBlockFetchClient :: !Bool
```
Comment (Jimbo4350) [inline cardano-config/src/Cardano/Config/Types.hs]: This is fragile. Currently if you add a tracer, you would need to make sure that 'NodeConfiguration`'s `FromJSON` instance "aligns" with `TraceOptions`.

Potential solution: Create a data type isomorphic to bool for each tracer. I'd prefer a more elegant solution to this.

Comment (rvl) [reply cardano-config/src/Cardano/Config/Types.hs]: If you split this type, you should be able to use Generic FromJSON, and have a nicer config format. Something like this:

```haskell
data TraceOptions = TraceOptions
  { traceVerbosity :: !TracingVerbosity
  , traceEnable :: !TraceEnable
  }
data TraceEnable = TraceEnable
  { traceBlockFetchClient :: !Bool
  , traceBlockFetchDecisions :: !Bool
  , traceBlockFetchProtocol :: !Bool
  , ...
  }
```

Then the YAML file could look like:
```yaml
tracing:
  verbosity: brief
  enable:
  - BlockFetchClient: false
  - BlockFetchDecisions: false
  - BlockFetchProtocol: false
  - ...
```

Comment (Jimbo4350) [reply cardano-config/src/Cardano/Config/Types.hs]: @rvl the plan so far seems to be having the tracers formatted in the yaml file as follows:
(https://github.com/input-output-hk/cardano-node/issues/443#issuecomment-569678510)

```
TraceBlockFetchClient:
    Enabled: True
    MinSeverity: Debug
    Verbosity: MaximumVerbosity
```
Splitting `TraceOptions` doesn't alleviate the potential to mix up of tracers if the `FromJSON` instance of `NodeConfiguration` and `TraceOptions` records aren't aligned.

**Comment by Jimbo4350** [comment]
Failing with:
```
machine# [   16.724792] bc50mcgyx2rxcw6a1n4dngsvp8mknp48-unit-script-cardano-node_-start[1183]: Error in $: key "TracingVerbosity" not present
```
Need to update the generation of the config `.yaml` files in the nix chairman cluster test.

**Comment by jbgi** [comment]
https://github.com/input-output-hk/cardano-node/commit/95d9fb239ef4548149ef157d2163d3a10ca7a610 should probably be reverted as part of this PR.

File: nix/nixos/cardano-node-service.nix [hunk 2]
```diff
@@ -25,11 +25,6 @@ let
             "--signing-key ${cfg.signingKey}"}"
           "${lib.optionalString (cfg.delegationCertificate != null)
             "--delegation-certificate ${cfg.delegationCertificate}"}"
-          "--tracing-verbosity-${cfg.tracingVerbosity}"
```
Comment (jbgi) [inline nix/nixos/cardano-node-service.nix]: the `tracingVerbosity` option definition could be removed too.

Comment (Jimbo4350) [reply nix/nixos/cardano-node-service.nix]: @jbgi removed. This should pass ci now. Once it does are we good to merge: https://github.com/input-output-hk/iohk-nix/pull/303

**Comment by Jimbo4350** [comment]
Don't merge yet. I need to re-point `iohk-nix` at `master`.

**Pull request closed without merging**